src/controllers/url.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from 'express';
+import { customAlphabet } from 'nanoid';
+import Url from '../models/url';
+import { BASE_URL, HTTP_ERRORS } from '../constant';
+import logger from '../logger/index';
+
+const createShortUrls = async (req: Request, res: Response) => {
+  try {
+    const { url, customName } = req.body;
+    let shortUrls;
+    const generateShortId = customAlphabet(
+      'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890',
+      5
+    );
+
+    if (!url) {
+      return res.status(HTTP_ERRORS.BAD_REQUEST).json({ error: 'Original Url is required' });
+    }
+    const existingLongUrl = await Url.findOne({ originalUrl: url });
+    if (existingLongUrl) {
+      return res
+        .status(200)
+        .json({ message: 'Shortened url already exists', shorturl: existingLongUrl.shortUrl });
+    }
+
+    if (customName) {
+      const trimmedName = customName.split(' ').join('');
+      if (trimmedName.length < 5) {
+        return res
+          .status(HTTP_ERRORS.BAD_REQUEST)
+          .json({ error: 'Custom name must be at least five letters' });
+      }
+      const customNamewithDashes = customName.split(' ').join('-');
+      shortUrls = `${BASE_URL}/${customNamewithDashes}`;
+      const existingCustomName = await Url.findOne({ customName });
```

Comment:
You are using `findOne` twice which is not performant. You should find once and share the result